### DISCUSSION TITLE
Langsmith and langserve compatibility

### DISCUSSION THREAD

**Discussion by Blubberblub** [langchain-ai/langsmith-sdk#547] [Q&A] [answered]
We have a web front-end, that uses a `RemoteRunnable`  with `streamEvents` from langserve. We want to add user feedback that gets send to langsmith from our frontend. Therefore we tried to get the run_id through a `RunCollectorCallbackHandler ` as described in the docs to get the run_id from the `remoteChain.streamEvents()`. Is there other ways or is the bug on the langserve side?
Docs reference for RunCollectorCallbackHandler :
https://docs.smith.langchain.com/tracing/faq/langchain_specific_guides#how-do-i-change-the-tracer-project

**Comment by hinthornw** [comment]
We recently added support to direclty specify it in the runnable config. That's probably the most convenient way:

```
run_id = uuid.uuid4()
chain.invoke(..., {"run_id": run_id})

```

The collector is only tirggered at the end of streaming, which is a limitation

**Reply by Blubberblub** [reply, marked as answer]
@hinthornw Thanks for the reply. Interesting idea to set it when invoking the chain from the front-end, i did not know that was possible. I am retrieving the run_id from the event logs using `streamEvents` on the `RemoteRunnable` now. There was a small issue where the `streamEvents` function did not return the run_ids. It is now solved by the langchainjs maintainers, so everything is fine.
